### PR TITLE
Release a lock that is no longer used

### PR DESCRIPTION
**Pull request by ycaibb**
Release a lock threadcount_mutex that is no longer used

Signed-off-by: ycaibb <ycaibb@gmail.com>

### CODE REVIEW

**Comment by grondo** [comment]
Can you more explicitly describe the problem in the commit message, e.g. something like:

>  In dsh(), the threadcount_mutex is never released. While technically not a bug, this could cause a problem or confusion in the future if a lock is attempted on the mutex.

Then describe the fix. I think this will help future developers when looking at project history.

**Comment by ycaibb** [comment]
@grondo  Hi, I have revised the commit message.

File: src/pdsh/dsh.c [hunk 1]
```diff
@@ -1192,7 +1192,8 @@ int dsh(opt_t * opt)
     dsh_mutex_lock(&threadcount_mutex);
     while (threadcount > 0)
         pthread_cond_wait(&threadcount_cond, &threadcount_mutex);
-
+    dsh_mutex_unlock(&threadcount_mutex);
+
```
Comment (grondo) [inline src/pdsh/dsh.c:1196]: Can you fix this whitespace error?
```suggestion

```

Comment (grondo) [reply src/pdsh/dsh.c:1196]: Also, please wrap commit message body at 72 columns when possible.
A description of why (and what makes a good commit message) can be found [here](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)

**Review by grondo — CHANGES_REQUESTED** [review]
Thanks! I've noted a couple requested minor changes inline.

File: src/pdsh/dsh.c (hunk 1 above)
Comment (ycaibb) [reply src/pdsh/dsh.c]: Please take a look, Thanks~

**Review by grondo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**